template.go

```diff
@@ -4,36 +4,97 @@ import (
 	"bytes"
 	"encoding/json"
 	"errors"
+	"fmt"
 )
 
 // Template represents a Template for launching instances from
 type Template struct {
 	ID               string `json:"id"`
 	Code             string `json:"code"`
 	Name             string `json:"name"`
-	AccountID        string `json:"account_id"`
-	ImageID          string `json:"image_id"`
+	AccountID        string `json:"account_id,omitempty"`
+	ImageID          string `json:"image_id,omitempty"`
 	VolumeID         string `json:"volume_id"`
 	ShortDescription string `json:"short_description"`
 	Description      string `json:"description"`
 	DefaultUsername  string `json:"default_username"`
 	CloudConfig      string `json:"cloud_config"`
 }
 
-// GetTemplateByCode finds the Template for an account with the specified code
-func (c *Client) GetTemplateByCode(code string) (*Template, error) {
+// ListTemplates return all template in system
+func (c *Client) ListTemplates() ([]Template, error) {
```

Comment:
Why have you replaced `GetTemplateByCode`? Having a `listTemplates` method is good, but the other function is a useful one still.

Reply:
Never mind, I see it's just diff weirdness - it's further down the page.